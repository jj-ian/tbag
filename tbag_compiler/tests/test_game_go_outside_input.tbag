/* Authors: Julie */
room {}

room Closet { 
	name = "Closet";
}

room Bedroom {
	name = "Bedroom";
}


room LivingRoom {
	name = "Living Room";
}

room Outside {
	name = "Outside";
}

Closet <-> Bedroom;
Bedroom <-> LivingRoom;

start { Bedroom }

boolean madeItOutside = false;

currentRoom == Bedroom {
	print("Currently in: ");
	print(currentRoom.name);
	print("\n");
	getInputFromOptions("Closet", "LivingRoom");
	->input
	print("\n");
}

currentRoom == Closet {
	string testName = currentRoom.name;
	print("Currently in: ");
	print(testName);
	print("\n");
	getInputFromOptions("Bedroom");
	->input
	print("\n");
}

currentRoom == LivingRoom {
	print("Currently in: ");
	print(currentRoom.name);
	print("\n");
	getInputFromOptions("Bedroom", "Outside");
	->input
	print("\n");
}

currentRoom == Outside {
	print("Currently in: ");
	print(currentRoom.name);
	print("\n");
	print("\n");
	madeItOutside = true;
}

madeItOutside {
	print("Good job making it outside, lazybones.");
	print("\n");
	endgame;
}
