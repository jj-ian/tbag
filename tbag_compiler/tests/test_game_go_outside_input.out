Currently in: Bedroom
Choose from one of the following options:
Closet
LivingRoom

Currently in: Living Room
Choose from one of the following options:
Bedroom
Outside

Currently in: Outside

Good job making it outside, lazybones.
